src/Aspire.Dashboard/Otlp/WebApplicationExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using Aspire.Dashboard.Authentication;
+using Aspire.Dashboard.Otlp.Grpc;
+using Microsoft.AspNetCore.Mvc;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp;
+
+public static class WebApplicationExtensions
+{
+    // https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/large-object-heap
+    private const int MaxSizeLessThanLOH = 84999;
+
+    public static WebApplication ConfigureHttpOtlp(this WebApplication app, Uri httpEndpoint)
+    {
+        app.MapPost("/v1/metrics",
+                async ([FromServices] OtlpMetricsService service, HttpContext httpContext) =>
```

Comment:
Is this injecting the OTLP gRPC service into the OTLP HTTP API? I haven't looked closely into what this means, but on the surface I'm not sure I like that coupling.

Reply:
I'm with you. I would like to have a protocol indipendent service to export otlp data